src/rez/cli/rm.py

```diff
@@ -46,6 +52,27 @@ def remove_package(opts, parser):
         sys.exit(1)
 
 
+def remove_package_family(opts, parser):
+    from rez.vendor.version.requirement import VersionedObject
+    from rez.package_remove import remove_package_family
+
+    if opts.dry_run:
+        parser.error("--dry-run is not supported with --family")
```

Comment:
dry run could probably at least tell you if it would allow proceeding with an archival if it _weren't_ a dry run?